app/services/registration_service.rb

```diff
@@ -0,0 +1,9 @@
+class RegistrationService
+  def from_invitation(email, invitation_token, user_data = {})
+    Invitation.find_by(email: email, token: invitation_token)
```

Comment:
mongoid raises an error if element not found. it can be configured but I prefer (for now) not to change the default behaviour. [mongo docs](http://mongoid.org/en/mongoid/docs/querying.html)
Answering your questions.
* I pretend a user can't be registered unless it has received an invitation.
* Yes, it's like `return unless invitation = Invitation.find_by(....)`, but mongoid raises exception by default as I mention above.